legacy-migration/providers/migrate.py

```diff
@@ -0,0 +1,312 @@
+"""Migrates providers from legacy services REST API into new providers interface"""
+import json
+import os
+import xml.etree.ElementTree as ET
+import re
+import time
+import sys
+import requests
+
+URL_ROOT = "https://cmr.sit.earthdata.nasa.gov"
+
+access_token = os.environ.get('SIT_SYSTEM_TOKEN')
+CMR_INGEST_ENDPOINT = "https://cmr.sit.earthdata.nasa.gov/ingest"
+
+# To test local ingesting into CMR
+local_cmr_access_token = os.environ.get('localHeader')
+# These are for the Enum values that we must set in the metadata field
+# TODO I want more clarity on how to handle these organization and contact roles since there is not an enum accessible substitute in the old format
+DEFAULT_ORGANIZATION_ROLE = "SERVICE PROVIDER"
+DEFAULT_CONTACT_ROLE = "TECHNICAL CONTACT"
+
+# Metadata specification for ingesting providers
+# It should not be expected this version or schema directories will change
+# within the lifecycle of this migration
+metadata_specification = {
+            "Name": "provider",
+            "Version": "1.0.0",
+            "URL": "https://cdn.earthdata.nasa.gov/schemas/provider/v1.0.0"}
+
+def ingest_provider(provider_metadata, provider_id):
+    """Attempts to ingest metadata into CMR (locally)"""
+    response = requests.post("http://localhost:3002/providers/",
+                        data=provider_metadata,
+                        headers={"Authorization": local_cmr_access_token, "User-id": "legacy_migration",
+                        "Content-Type": "application/json"},
+                        timeout=60
+                        )
+    if response.status_code >= 300:
+        print("Failed to ingest a record for the provider : ", provider_id)
+        ingest_error_log = open("./logs/"+str(provider_id)+"_error","w" ,encoding="utf8")
+        ingest_error_log.write('Error for provider ' + provider_id + ' ingesting ' + 'due to ' +
+        str(response.content))
+    else:
+        print("Successfully Ingested metadata into CMR ", response.status_code)
+    return response.content
+
+def is_admin_group(group):
+    """Looks for admin in the name of the group"""
+    # if in the name of the text admin comes up anywhere then it is an admin group
+    return bool(re.search('admin', group.lower()))
+
+def get_admin_usernames(member_guid_arr):
+    """Parses the list of member_guids and retrieves the edl usernames though an API call"""
+    username_arr = []
+    if len(member_guid_arr) < 1:
+        return username_arr
+    #TODO: can we do this in one request instead of iterating over each username
+    for user_id in member_guid_arr:
+        print("User_id being passed in request", user_id)
+        try:
+            response = requests.get(f"{URL_ROOT}/legacy-services/rest/users/{user_id}",
```

Comment:
URL_ROOT is just for legacy services or should this also be for all CMR?

Reply:
CMR_INGEST_ENDPOINT is a whole URL, but here your just a partial, I like the partial approach, if the prefix is the same for both, then both should use the same var

Reply:
They're the same that is a good point I'll reuse it